bin/infection

```diff
@@ -9,6 +9,10 @@ $autoloaderInWorkingDirectory = getcwd() . '/vendor/autoload.php';
 
 if (is_file($autoloaderInWorkingDirectory)) {
     require_once $autoloaderInWorkingDirectory;
+} elseif ($vendorDir = exec('composer config vendor-dir')) {
+    if (is_file(getcwd() . '/' . $vendorDir . '/autoload.php')) {
+        require_once getcwd() . '/' . $vendorDir . '/autoload.php';
+    }
```

Comment:
Below this line there is the place where we try to guess the path to `autoload.php`. If you do ask composer for the vendor-dir here, then that guessing below is almost never going to be used.

So I think for now best would be to use this new measure as the last effort measure, when everything else fails. Otherwise put, this better be right before that `if (!isset($file)) { fwrite(STDERR, ...`.

Also, it does not make sense to run `composer config` if there's no `composer.json` present.

Reply:
While I'm running test __DIR__ constant (used in $files) and getcwd() (which used in $autoloaderInWorkingDirectory) is not equal. First one install classmap from infection/composer.json (that included infection sources) and the last one - from composer.json that located in my e2e test directory and contains only test source and phpunit. Maybe you can help me - I don't fully understand what's goes on production enviroment